lib/BC/Util.cpp

```diff
@@ -1670,20 +1673,13 @@ void CloneFunctionInto(llvm::Function *source_func, llvm::Function *dest_func,
 
       auto new_inst = old_inst.clone();
 
-      // Resetthe metadata after cloning.
+      // Reset the metadata after cloning.
       for (auto [md_id, val] : mds) {
         old_inst.setMetadata(md_id, val);
       }
 
-      // NOTE(pag): This is pretty evil, there's no reliable way to move the
-      //            type to the destination context, and there are assertions,
-      //            e.g. in `llvm::CallBase::setCalledFunction`, that
-      //            (correctly) detect that the types don't match.
-      new_inst->*REMILL_ACCESS_MEMBER(llvm, Value, VTy) =
-          RecontextualizeType(new_inst->getType(), dest_context, type_map);
-
+      new_block_builder.Insert(new_inst);
```

Comment:
I should be able to do this without constructing an `IRBuilder`, lemme just find the right API.